pkg/apis/submariner.io/v1/types.go

```diff
@@ -271,3 +272,64 @@ type ClusterGlobalEgressIPList struct {
 
 	Items []ClusterGlobalEgressIP `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GlobalIngressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of desired behavior of GlobalIngressIP object.
+	Spec GlobalIngressIPSpec `json:"spec"`
+
+	// Observed status of GlobalIngressIP. Its a read-only field.
+	// +optional
+	Status GlobalIngressIPStatus `json:"status,omitempty"`
+}
+
+type GlobalIngressIPSpec struct {
+	// TargetType can be ClusterIPService or a Pod
+	Target TargetType `json:"target"`
+
+	// If the TargetType points to a Pod that belongs to a Headless Service, it uses the
+	// name of the corresponding Service.
+	// +Optional
+	ServiceName string `json:"serviceName,omitempty"`
+
+	// Since the Name of GlobalIngressIP will be prepended with "svc-" or "pod-" based on
+	// the object, we may have to truncate the name in some cases when the length exceeds the
+	// maximum length that is allowed by K8s. In such cases, the objRef will include the
+	// full name of the object.
+	// +Optional
+	ObjRef *corev1.LocalObjectReference `json:"objRef,omitempty"`
```

Comment:
I think we should make this mandatory (ie non-optional) to simplify.

```suggestion
	// The reference to the Target object. 
	TargetRef corev1.LocalObjectReference `json:"targetRef,omitempty"`
```

Reply:
omitempty can be removed then?

Reply:
yeah we might as well remove it - it doesn't make sense for a non-pointer struct as it has no empty value.